File: server/src/main/java/cn/iinti/atom/system/AppContext.java
package cn.iinti.atom.system;


import cn.iinti.atom.entity.UserInfo;
import jakarta.servlet.http.HttpServletResponse;
import org.apache.commons.lang3.BooleanUtils;

public class AppContext {
    private static final ThreadLocal<UserInfo> userInfoThreadLocal = new ThreadLocal<>();

    private static final ThreadLocal<Boolean> API_USER = new ThreadLocal<>();

    private static final ThreadLocal<LoginRequired> LOGIN_ANNOTATION = new ThreadLocal<>();
    private static final ThreadLocal<HttpServletResponse> SERVLET_RESPONSE = new ThreadLocal<>();

    public static UserInfo getUser() {
        return userInfoThreadLocal.get();
    }

    public static void setUser(UserInfo user) {
        userInfoThreadLocal.set(user);
    }

    public static void markApiUser() {
        API_USER.set(true);
    }

    public static void setLoginAnnotation(LoginRequired loginRequired) {
        LOGIN_ANNOTATION.set(loginRequired);
    }

    public static LoginRequired getLoginAnnotation() {
        return LOGIN_ANNOTATION.get();
    }

    public static HttpServletResponse getServletResponse() {
        return SERVLET_RESPONSE.get();
    }

    public static boolean isApiUser() {
        Boolean aBoolean = API_USER.get();
        return BooleanUtils.isTrue(aBoolean);
    }

    public static void removeUser() {
        userInfoThreadLocal.remove();
        API_USER.remove();
        LOGIN_ANNOTATION.remove();
        SERVLET_RESPONSE.remove();
    }
}

File: server/src/main/kotlin/cn/iinti/atom/system/AppContext.kt
package cn.iinti.atom.system

import cn.iinti.atom.entity.UserInfo
import jakarta.servlet.http.HttpServletResponse
import org.apache.commons.lang3.BooleanUtils

object AppContext {
    private val userInfoThreadLocal = ThreadLocal<UserInfo>()
    private val API_USER = ThreadLocal<Boolean>()
    private val LOGIN_ANNOTATION = ThreadLocal<LoginRequired>()
    private val SERVLET_RESPONSE = ThreadLocal<HttpServletResponse>()

    @JvmStatic
    fun getUser(): UserInfo {
        return userInfoThreadLocal.get()
    }

    fun setUser(user: UserInfo?) {
        userInfoThreadLocal.set(user)
    }

    fun markApiUser() {
        API_USER.set(true)
    }

    fun setLoginAnnotation(loginRequired: LoginRequired?) {
        LOGIN_ANNOTATION.set(loginRequired)
    }

    fun getLoginAnnotation(): LoginRequired? {
        return LOGIN_ANNOTATION.get()
    }

    fun getServletResponse(): HttpServletResponse? {
        return SERVLET_RESPONSE.get()
    }

    fun isApiUser(): Boolean {
        val aBoolean = API_USER.get()
        return BooleanUtils.isTrue(aBoolean)
    }

    fun removeUser() {
        userInfoThreadLocal.remove()
        API_USER.remove()
        LOGIN_ANNOTATION.remove()
        SERVLET_RESPONSE.remove()
    }
}